### DISCUSSION TITLE
GeoLocation unable to start Listener on IOS 16.0

### DISCUSSION THREAD

**Discussion by dinonovak** [shinyorg/shiny#1110] [Q&A] [answered]
I am trying to initialise GPS, in Xamarin Forms App (on IOS 16.0 in simulator)
Forms 5.0.0.2515
Shiny.Locations & Shiny.notifivations 2.5.3 
Location tracking set to Always in App Settings

```
                    var result = await this.gpsmanager.RequestAccess(new GpsRequest { BackgroundMode = GpsBackgroundMode.Standard });

                    if (result != AccessState.Available)
                    {
                        await this.Alert("Insufifficient permissions!");
                        return;
                    }

                    var request = new GpsRequest
                    {
                        Accuracy = GpsAccuracy.Normal,
                        BackgroundMode = GpsBackgroundMode.Realtime
                    };

                    try
                    {
                        await this.gpsmanager.StartListener(request);
                    }
                    catch (Exception ex)
                    {
                        await this.Alert(ex.ToString());
                    }
```

But when reaching await this.gpsmanager.StartListener(request);
I am getting following error:
System.ArgumentException: Failed to add secure value - MissingEntitlement   at Shiny.Stores.SecureKeyValueStore.Set (System.String key, System.Object value) [0x0006f] in <6e9f3393ab8a466ca84d7ca292cbbba2>:0    at Shiny.StoreExtensions.SetOrRemove (Shiny.Stores…

Anyone has idea what could be the problem?

I have  permission set in info.plist
```
	<key>NSMotionUsageDescription</key>
	<string>Required for pedometer</string>
	<key>NSLocationAlwaysUsageDescription</key>
	<string>The beacons or geofences or GPS always have you!</string>
	<key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
	<string>The beacons or geofences or GPS always have you!</string>
	<key>NSLocationWhenInUseUsageDescription</key>
	<string>The beacons or geofences or GPS always have you!</string>
	<key>UIBackgroundModes</key>
	<array>
		<string>location</string>
	</array>
```

**Comment by aritchie** [comment, marked as answer]
Secure store is used under the hood to make sure GPS auto-starts again until your code turns it off.  If you are using anything with shiny in the simulator, you have to add the keychain entitlement and make sure your build includes it.

**Comment by dinonovak** [comment]
Updated entitlements.plist and included in build so now all works!
